src/helpers/single-value-providers.ts

```diff
@@ -50,6 +52,25 @@ export class CustomFieldIntegerValueProvider extends SingleValueProvider<number>
     }
 }
 
+export class DaysWithDataValueProvider extends SingleValueProvider<number> {
+    constructor(public dataType: string, public daysInPast: number) {
+        if(daysInPast < 0) {
+            throw new Error('daysInPast must be a non-negative number');
+        }
+        super('days with data');
+    }
+
+    async getValue(): Promise<number | undefined> {
+        const today = new Date();
+        const startDate = startOfDay(subDays(today, this.daysInPast));
```

Comment:
I'm not sure it is 100% necessary, but you may want to ensure that both the start and end dates are "start of day" dates.  Maybe something like this:

```
const today = startOfToday();
const startDate = subDays(today, this.daysInPast);
```